var fs = require("fs");

// fs.readFile('input.txt', function (err, data) {
//    if (err) return console.error(err);
//    console.log(data.toString());
// });

// var myCallback = function(err, data) {
//     if (err) throw err; // Check for the error and throw if it exists.
//     console.log('got data: '+data); // Otherwise proceed as usual.
//   };
//   var usingItNow = function(callback) {
//     var myError = new Error('My custom error!');
//     callback(null, 'get it?'); // I send my error as the first argument.
//   };
//   usingItNow(myCallback);

// console.log("Program Ended");


var data
  var usingItNow = function (function(err,data) {
    var myError = new Error('My custom error!');
    function (err, data) {
      if (err) throw err; 
      console.log('got data: '+data); 
    }; )
  };
  usingItNow('err','xxx');

console.log("Program Ended");